monkey/infection_monkey/monkey.py

```diff
@@ -295,7 +296,8 @@ def _build_server_list(self, relay_port: int) -> Sequence[str]:
         return list(ordered_servers.keys())
 
     def _build_puppet(self) -> IPuppet:
-        puppet = Puppet(self._agent_event_queue)
+        plugin_registry = PluginRegistry()
+        puppet = Puppet(self._agent_event_queue, plugin_registry)
 
         puppet.load_plugin(
```

Comment:
I don't understand why it's the responsibility of the puppet to load the plugin? Puppet doesn't store plugins, so it shouldn't have `load_plugin` method 🤔 What we should do is load the plugins into the registry and then pass the registry to the puppet

Reply:
It'll go away eventually, but just leave it for now. It's not worth changing.